lua-tree/share/lua/5.1/kong/tools/utils.lua

```diff
@@ -1454,16 +1257,12 @@ do
   end
 end
 
-local time_ns
-do
-  local nanop = ffi_new("nanotime[1]")
-  function time_ns()
-    -- CLOCK_REALTIME -> 0
-    C.clock_gettime(0, nanop)
-    local t = nanop[0]
+-- No reason to yield a Lua coroutine based on `ngx.get_phase()`,
+-- as we're not running this within Nginx anyway.
+_M.yield = function() return end
 
-    return tonumber(t.tv_sec) * 1e9 + tonumber(t.tv_nsec)
-  end
+local time_ns = function()
+    return require("os").time() * 10^9
```

Comment:
The gopher-lua VM does support this & returns the current Unix epoch in seconds, as expected:
https://github.com/yuin/gopher-lua/blob/658193537a640772633e656f4673334fe1644944/oslib.go#L67

Reply:
Will defer this conversation to [this thread](https://github.com/Kong/goks/pull/27#discussion_r958517611).